specs/chugsplash-manager.md

```diff
@@ -0,0 +1,76 @@
+# ChugSplashManager Speed Spec
+
+## Bundle states
+
+```typescript
+enum BundleStatus {
+  EMPTY,
+  PROPOSED,
+  APPROVED,
+  CANCELLED,
+  COMPLETED
+}
+```
+
+## Computing a unique bundle ID
+
+```typescript
+const computeBundleId = (
+  bundleRoot: bytes32,
+  bundleSize: uint256,
+  configUri: string
+): bytes => {
+  return keccak256(
+    abi.encode(
+      bundleRoot,
+      bundleSize,
+      configUri,
+    )
+  )
+}
+```
+
+## Proposing a ChugSplash bundle
+
+* A ChugSplash bundle MUST only be proposable by the `admin` of the `ChugSplashManager`.
+* Bundles MUST be in the `EMPTY` state before they can be proposed.
```

Comment:
This prevents users from being able to roll back a project to a previously executed bundle. I think we can fix this by just changing this to "Bundles MUST NOT be in the `PROPOSED` or `APPROVED` state before they can be proposed."

Reply:
Let's ignore rollbacks for the v1, there's a lot to figure out with how to do rollbacks safely